deluge/common.py

```diff
@@ -962,23 +995,69 @@ def is_ipv6(ip):
     """
 
     try:
-        import ipaddress
-    except ImportError:
-        import socket
+        socket.inet_pton(socket.AF_INET6, ip)
+    except OSError:
+        return False
+    else:
+        return True
+
+
+def is_adapter(interface):
+    """A test to see if 'interface' is a valid network interface.
+
+    Args:
+        interface (str): The Interface to test. eg. eth0 linux. GUID on Windows.
+
+    Returns:
+        bool: Whether Interface is valid or not.
+
+    Examples:
+        >>> is_adapter("eth0")
+        True
+        >>> is_adapter("{7A30AE62-23ZA-3744-Z844-A5B042524871}")
+        True
 
+    """
+
+    if linux_check() or osx_check():
```

Comment:
Can we put ifaddr check first.

Also with an OSError we are unable to determine adapter status so have to return True.

We could include a fallback for Windows with your regex to do a pattern match 

So something like:
```python
if ifaddr:
    try:
        adapters = ifaddr.get_adapters()
    except OSError:
         return True
    else:
         return any([adapter == a.name for a in adapters])

if linux_check() or osx_check():
    try:
           adapters = socket.if_nameindex()
    except OSError:
         return True
    else:
         return any([adapter == a[1] for a in adapters])

if windows_check():
     regex = '^{[0-9A-Z]{8}-([0-9A-Z]{4}-){3}[0-9A-Z]{12}}$'
     return  re.search(regex, str(interface))

return True

Reply:
Would it not be better to reply first on built-in vs third party? I would think third party would be less reliable, which is why I have it only using that on windows because windows we dont have a built-in, at least not yet.

Reply:
Okay, there are pros and cons to third party libs but I'll concede here that we could put ifaddr after. I've updated the snippet to also pass on OSError so that it tries `ifaddr` if it's available and also rather than testing for linux/macos we just do negative check for windows to exclude it